### PR TITLE
hack together support for putting NULL fields in by default

### PR DESCRIPTION
**Pull request by osheroff**
@zendesk/rules

### CODE REVIEW

**Comment by dadah89** [comment]
failing tests

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 1]
```diff
@@ -212,6 +214,12 @@ private void parse(String [] argv) {
 		if ( options.has("replay")) {
 			this.replayMode = true;
 		}
+
+		if ( properties.contains("omit_nulls") )
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: fetchOption?

File: src/main/java/com/zendesk/maxwell/RowMap.java [hunk 2]
```diff
@@ -215,7 +217,7 @@ public String toJSON() throws IOException {
 			}
 		}
 
-		writeMapToJSON("data", this.data, false);
+		writeMapToJSON("data", this.data, !this.omitNulls);
 
 		if ( !this.oldData.isEmpty() ) {
 			writeMapToJSON("old", this.oldData, true);
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/RowMap.java]: missed passing the flag here

File: src/main/java/com/zendesk/maxwell/schema/columndef/EnumColumnDef.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ private String asString(Object value) {
 		Integer i = (Integer) value;
 
 		if ( i == 0 )
-			return null;
+			return "";
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/schema/columndef/EnumColumnDef.java]: why was this necessary?

File: src/test/java/com/zendesk/maxwell/RowMapBufferTest.java [hunk 4]
```diff
@@ -13,9 +13,9 @@ public void TestOverflowToDisk() throws Exception {
 		RowMapBuffer buffer = new RowMapBuffer(2, 250); // allow about 250 bytes of memory to be used
 
 		RowMap r;
-		buffer.add(new RowMap("insert", "foo", "bar", 1L, new ArrayList<String>(), new BinlogPosition(3, "mysql.1")));
-		buffer.add(new RowMap("insert", "foo", "bar", 2L, new ArrayList<String>(), new BinlogPosition(3, "mysql.1")));
-		buffer.add(new RowMap("insert", "foo", "bar", 3L, new ArrayList<String>(), new BinlogPosition(3, "mysql.1")));
+		buffer.add(new RowMap("insert", "foo", "bar", 1L, new ArrayList<String>(), new BinlogPosition(3, "mysql.1"), true));
```
Comment (dadah89) [inline src/test/java/com/zendesk/maxwell/RowMapBufferTest.java]: suppose you need a fixture change (expected result) for this to pass?

File: src/main/java/com/zendesk/maxwell/schema/columndef/EnumColumnDef.java (hunk 3 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/schema/columndef/EnumColumnDef.java]: this emulates what mysql does when you pass it an invalid enum; it's stored internally as a "0" but when SELECTed it appears as the empty string.

File: src/main/java/com/zendesk/maxwell/RowMap.java (hunk 2 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/RowMap.java]: that's intentional: `oldData` only contains data that's changed in the row, and we *always* put a null value in there if the row was null.

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: would have to write a version of `fetchBooleanOption`, which is possible.

**Comment by xmlking** [comment]
Any chance this PR will  be released in near time ?
Thanks

**Comment by osheroff** [comment]
@xmlking yeah @dadah89 did a cleaner PR around the area of controlling the row format.  I re-coded for that work and pushed here.

**Comment by osheroff** [comment]
@dadah89 code-review here at your leisure

**Comment by xmlking** [comment]
I thing  `output_null=true` is more readable then  `output_omit_null=false`

**Comment by dadah89** [comment]
changing the flag name makes sense to me, up to you
lgtm 👍

**Comment by osheroff** [comment]
ok @xmlking good suggestion, updated

**Comment by osheroff** [comment]
@dadah89 one last CR?

**Comment by dadah89** [comment]
👍 !
